### PR TITLE
Eucledian distance

### PR DESCRIPTION
**Pull request by audhalbritter**
Calculate euclidean (had to check spelling) distance in each turf over time (2012 - 2016) for div/conv.
And make a plot.

### CODE REVIEW

File: R/euclidean_distance_plan.R [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  fat_table = bind_rows(
+    convergence = sum_boot_moment_conv,
+    divergence = sum_boot_moment_div, 
+    .id = "direction") %>%
+    ungroup() %>% 
+    select(direction, Site, blockID, trait, TTtreat, year, turfID, Mean) %>% 
+    pivot_wider(names_from = trait, values_from = Mean) %>% 
+    filter(!is.na(C_percent),
+           year %in% c(2012, 2016),
+           !c(direction == "convergence" & Site == "L"),
```
Comment (richardjtelford) [inline R/euclidean_distance_plan.R:14]: you don't need `c` here, just the brackets

File: R/euclidean_distance_plan.R [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    rowid_to_column(),
+  
+  #select only traits
+  only_traits <- fat_table %>% 
+    ungroup() %>% 
+    select(C_percent:Wet_Mass_g),
+  
+  #scale, calculate euclidean distance
+  d = dist(scale(only_traits)),
+  D = as.matrix(d),
```
Comment (richardjtelford) [inline R/euclidean_distance_plan.R]: merge d and D into one target - probably also only_traits and maybe distances.

File: R/euclidean_distance_plan.R [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  distances = tibble(row = as.vector(row(D)), 
+                      col = as.vector(col(D)), 
+                      dist = as.vector(D)) %>% 
+    filter(row > col) %>% # one half of the matrix
+    # join meta data by row and col
+    left_join(fat_meta, by = c("row" = "rowid")) %>% 
+    left_join(fat_meta, by = c("col" = "rowid"), suffix = c(".row", ".col")) %>% 
+    #sfilter distances within the same block
+    filter(blockID.row == blockID.col) %>% # not sure if this step is needed...?
```
Comment (richardjtelford) [inline R/euclidean_distance_plan.R:38]: probably redundant as 42 will enforce this anyway

File: R/plots_plan.R [hunk 4]
```diff
@@ -19,6 +19,18 @@ plot_plan <- drake_plan(
     facet_wrap(~trait, scales = "free_y"),
   
 
+  #H1Q2 + 3: Conv/div along gradient and across treatments
+  Site <- c(H = "High Alpine",
```
Comment (richardjtelford) [inline R/plots_plan.R]: use = not <- in a plan

**Review by richardjtelford — CHANGES_REQUESTED** [review]

File: R/euclidean_distance_plan.R (hunk 3 above)
Comment (audhalbritter) [reply R/euclidean_distance_plan.R:38]: you mean line 39 is reduntant? That's what I tought yes...

File: R/plots_plan.R (hunk 4 above)
Comment (audhalbritter) [reply R/plots_plan.R]: I think I changed this in the next commit. will check

**Comment by audhalbritter** [comment]
have fixed your requests

**Review by richardjtelford — APPROVED** [review]

**Pull request merged**